test/generate_tests.py

```diff
@@ -156,3 +156,69 @@ def test_main_drizzle(self):
 			os.remove(test_cosmo_folder+'npy_files/img_%d.npy'%(i))
 		os.rmdir(test_cosmo_folder+'npy_files')
 		os.rmdir(output_folder)
+
+	def test_h5(self):
+		# Test that the h5 kwarg generates the required h5 files
+		old_sys = copy.deepcopy(sys.argv)
+		n_generate = 21
+		output_folder = 'test_data/test_dataset'
+		sys.argv = ['test','test_data/config_dict.py',output_folder,
+			'--n',str(n_generate),'--h5']
+		if tensorflow_installed:
+			sys.argv.append('--tf_record')
+		generate.main()
+
+		image_file_list = glob.glob(os.path.join(output_folder,'image_data.h5'))
+		# Check only 1 h5 file is made:
```

Comment:
There cannot be two files with the same name, so you're really just checking the file exists here. Maybe change the glob pattern to *.h5?